jbmc/src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -637,10 +637,8 @@ void java_bytecode_convert_classt::convert(
   typet field_type;
   if(f.signature.has_value())
   {
-    field_type=java_type_from_string_with_exception(
-      f.descriptor,
-      f.signature,
-      id2string(class_symbol.name));
+    field_type = *java_type_from_string_with_exception(
```

Comment:
Are we sure this has a value? (and other places)

Reply:
As above: no idea, but it was previously used unchecked. There are a couple of places that failed this while running the regression tests, and those I have now guarded appropriately. I still think that this makes `optionalt<typet>` much better as at least you'd have a failing assertion instead of silently passing around a `nil_typet` that some later stage won't know how to deal with.